code/Chapter4/Exercise_solutions/b_multiple_test_p_instant.cpp

```diff
@@ -0,0 +1,94 @@
+#include <gtest/gtest.h>
+
+// Define your function to be tested here.
+int Sum(int a, int b) {
+    return a + b;
+}
+
+// function to multiply two numbers
+double Multiply(double a, double b) {
+    return a * b;
+}
+
+// Function to calculate a raised to the power of b (integer) without using pow function
+double Power(double a, int b) {
+    double result = 1;
+    for(int i = 0; i < b; i++) {
+        result *= a;
+    }
+    return result;
+}
+
+// Define a test fixture class
+class ParameterizedTest : public testing::TestWithParam<std::pair<int, int>> {
+};
+
+// Define the test case with the parameterized test
+TEST_P(ParameterizedTest, TestSum) {
+    // Get the parameter values
+    int a = GetParam().first;
+    int b = GetParam().second;
```

Comment:
Again, you *could* use `std::tie`/structured bindings here, but it's also fine as is.

Reply:
In the previous examples also, we used this. So, I am willing to keep it. 

This gives us a chance to show more than one option of doing things. As you suggested, I have now added a better way in the last example and that should serve the purpose of what needs to be done in case the number of args increases.